src/Microsoft.ML.Data/Transforms/DropSlotsTransform.cs

```diff
@@ -331,40 +379,136 @@ private void GetSlotsMinMax(Column col, out int[] slotsMin, out int[] slotsMax)
+            for (int iinfo = 0; iinfo < slotsMin.Length; iinfo++)
+            {
+                var prevmax = -2;
+                for (int i = 0; i < slotsMin[iinfo].Length; i++)
+                {
+                    if (!(0 <= slotsMin[iinfo][i] && slotsMin[iinfo][i] < int.MaxValue))
+                        return false;
+                    if (!(0 <= slotsMax[iinfo][i] && slotsMax[iinfo][i] < int.MaxValue))
+                        return false;
+                    if (!(slotsMin[iinfo][i] <= slotsMax[iinfo][i]))
+                        return false;
+                    if (!(slotsMin[iinfo][i] - 1 > prevmax))
+                        return false;
+                    prevmax = slotsMax[iinfo][i];
+                }
+            }
+            return true;
+        }
+
+        protected override IRowMapper MakeRowMapper(Schema schema)
+            => new Mapper(this, schema);
+
+        private sealed class Mapper : OneToOneMapperBase
+        {
+            private readonly DropSlotsTransform _parent;
+            private readonly int[] _cols;
+            private readonly ColumnType[] _srcTypes;
+            private readonly ColumnType[] _dstTypes;
+            private readonly SlotDropper[] _slotDropper;
+            // Track if all the slots of the column are to be dropped.
+            private readonly bool[] _suppressed;
+            private readonly int[][] _categoricalRanges;
+
+            public Mapper(DropSlotsTransform parent, Schema inputSchema)
+                : base(parent.Host.Register(nameof(Mapper)), parent, inputSchema)
+            {
+                _parent = parent;
+                _cols = new int[_parent.ColumnPairs.Length];
+                _srcTypes = new ColumnType[_parent.ColumnPairs.Length];
+                _dstTypes = new ColumnType[_parent.ColumnPairs.Length];
+                _slotDropper = new SlotDropper[_parent.ColumnPairs.Length];
+                _suppressed = new bool[_parent.ColumnPairs.Length];
+                _categoricalRanges = new int[_parent.ColumnPairs.Length][];
+
+                for (int i = 0; i < _parent.ColumnPairs.Length; i++)
+                {
+                    if (!InputSchema.TryGetColumnIndex(_parent.ColumnPairs[i].input, out _cols[i]))
+                        throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", _parent.ColumnPairs[i].input);
+                    _srcTypes[i] = inputSchema.GetColumnType(_cols[i]);
+                    if (!IsValidColumnType(_srcTypes[i].ItemType))
+                        throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", _parent.ColumnPairs[i].input);
+                    _slotDropper[i] = new SlotDropper(_srcTypes[i].ValueCount, _parent.SlotsMin[i], _parent.SlotsMax[i]);
+                    ComputeType(inputSchema, i, _slotDropper[i], out _suppressed[i], out _dstTypes[i], out _categoricalRanges[i]);
+                }
+            }
+
+            private static bool IsValidColumnType(ColumnType type)
```

Comment:
Add a comment explaining what types are valid: scalar or vectors of string, float or double.